File: DSA/50_Days_50_Questions/724-find-pivot-index/find-pivot-index.py
class Solution(object):
    def pivotIndex(self, nums):
        n = len(nums)

        total = 0
        for i in range(n):
            total += nums[i]

        left = 0
        for i in range(n):
            right = total - left - nums[i]

            if(left == right ):
                return i
            
            left+=nums[i]
        
        return -1
        

File: DSA/50_Days_50_Questions/724-find-pivot-index/find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int n = nums.size();
        int total_sum = 0;
        for(int i=0; i<n; i++){
            total_sum +=nums[i];
        }

        int left = 0;
        for(int i=0; i<n; i++){
            int right = total_sum - left - nums[i];
            
            if(left == right){
                return i;
            }
            left += nums[i];
        }
        return -1;
    }
};